### ISSUE TITLE
CDN Consumption examples broken

### ISSUE THREAD

**Issue by ecyrb** [palantir/blueprint#6088]
#### Environment

The current (2023-04-22) CDN consumption example in the documentation website appears to be broken for me.

URL: https://blueprintjs.com/docs/#blueprint/getting-started.cdn-consumption
(Should be blueprint v4.18.0)

Chrome reports:
> Failed to load resource: the server responded with a status of 404 ()
> https://unpkg.com/@blueprintjs/core@4.18.0/dist/core.bundle.js

> Failed to load resource: the server responded with a status of 404 ()
> https://unpkg.com/@blueprintjs/icons@4.15.0/dist/icons.bundle.js

If I change the blueprint dependencies from:
```html
        <link href="https://unpkg.com/@blueprintjs/icons@^4.0.0/lib/css/blueprint-icons.css" rel="stylesheet" />
        <link href="https://unpkg.com/@blueprintjs/core@^4.0.0/lib/css/blueprint.css" rel="stylesheet" />
[...]
        <!-- Blueprint packages (note: packages must be topo-sorted, where dependencies come first) -->
        <script src="https://unpkg.com/@blueprintjs/icons@^4.0.0"></script>
        <script src="https://unpkg.com/@blueprintjs/core@^4.0.0"></script>

 ```
to `icons@4.11.0` and `core@4.13.0`, then it works.

#### Possible solution

Check `dist` output for NPM upload/publishing builds?

**Comment by ecyrb** [comment]
Seems to be fine for core@4.13.0 but not 4.14.0. Might be related to this commit? https://github.com/palantir/blueprint/commit/faf84d23de9a9bdc0e6af8e28fa752de5f8a99dc

**Comment by adidahiya** [comment]
We are generating the UMD bundle `core.bundle.js` and checking for its existence in our build (see the `assert-package-layout` script) but it seems like the `dist/` folder is being elided from the NPM tarball.

This was almost certainly caused by the upgrade to Node 16: https://github.com/palantir/blueprint/commit/aacc3c2a7b8435052fa62752b83f1760620e0a16

We've seen regressions in `npm publish` in newer versions of Node & NPM like this (https://github.com/palantir/blueprint/issues/5645) where certain folders just stop publishing seemingly at random.

I think the fix is to specify a `"files"` field in `package.json`. `npm publish` seems to respect those paths and always publish them correctly.

**Comment by adidahiya** [comment]
This is fixed in v5.0.0-alpha.1 with the addition of a "files" field to package.json (ec03c9101622e701b4fc451c4379b72f718dbe8e). I will cherry-pick that change onto develop.
